hppc/src/main/templates/com/carrotsearch/hppc/KTypeVTypeWormMap.java

```diff
@@ -0,0 +1,2011 @@
+            assert checkIndex(toIndex);
+        }
+        int offset = toIndex - fromIndex;
+        if (offset < 0) {
+            offset += next.length;
+        }
+        if (DEBUG_ENABLED) {
+            assert offset >= 0 : "offset=" + offset + ", fromIndex=" + fromIndex + ", toIndex=" + toIndex;
+        }
+        return offset;
+    }
+
+    /**
+     * Check the map integrity.
+     */
+    private boolean checkIntegrity(boolean checkSize) {
+        //noinspection ConstantConditions
+        assert DEBUG_ENABLED;
+        int entryCount = 0;
+
+        final KType[] keys = Intrinsics.<KType[]>cast(this.keys);
+        final VType[] values = Intrinsics.<VType[]>cast(this.values);
+
+        for (int index = 0; index < next.length; index++) {
+            // Check the distinction between empty and occupied bucket.
+            assert next[index] == 0 || (!Intrinsics.<KType>isEmpty(keys[index]) && values[index] != noValue());
+            // Check next[index] points to an occupied bucked.
+            assert next[index] == 0 || Math.abs(next[index]) == END_OF_CHAIN || next[addOffset(index, Math.abs(next[index]), next)] != 0;
+            // Check entry chains.
+            if (next[index] > 0) {
+                int i = index;
+                while (true) {
+                    // Check the hash of each entry in the chain is index.
+                    assert hash(keys[i], next.length) == index;
+                    if (Math.abs(next[i]) == END_OF_CHAIN) {
+                        break;
+                    }
+                    i = addOffset(i, Math.abs(next[i]), next); // Jump to the next entry in the chain.
+                    // Check tail-of-chain entries have negative next offset.
+                    assert next[i] < 0;
+                }
+            }
+            // Count occupied buckets.
+            if (next[index] != 0) {
+                entryCount++;
+            }
+        }
+        // Check map size.
+        assert !checkSize || entryCount == size;
+        return true;
+    }
+
+    private boolean checkIndex(int index) {
+        return checkIndex(index, next);
+    }
+
+    private static boolean checkIndex(int index, byte[] next) {
+        assert index >= 0 && index < next.length : "index=" + index + ", array length=" + next.length;
+        return true;
+    }
+
+    /**
+     * Efficient immutable set of excluded indexes (immutable int set of expected small size).
+     * <p/>Used when searching for a free bucket and attempting to move tail-of-chain entries
+     * recursively. We must not move the entry chains for which we want to find a free bucket.
+     * So {@link ExcludedIndexes} is immutable and can be stacked with {@link #union(ExcludedIndexes)}
+     * during recursive calls. In addition the initial {@link #NONE} is a constant and does not stack
+     * as it overrides {@link #union(ExcludedIndexes)}.
+     */
+    private static abstract class ExcludedIndexes {
```

Comment:
If it's a static class without any templated types then it could be pulled up so that it generates a single .class file (and not one for each template combination).

Reply:
I have moved to WormUtil all of the static classes and static methods that were not removed